docs/source/python/extending_types.rst

```diff
@@ -44,3 +44,162 @@ define the ``__arrow_array__`` method to return an Arrow array::
 
 The ``__arrow_array__`` method takes an optional `type` keyword which is passed
 through from :func:`pyarrow.array`.
+
```

Comment:
I realize this document doesn't talk about registration. Also show a simple IPC example?